core/modules/replica/sql/replication.sql

```diff
@@ -0,0 +1,759 @@
+SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 ;
+SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 ;
+SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='TRADITIONAL' ;
+
+-- --------------------------------------------------------------
+-- Table `config`
+-- --------------------------------------------------------------
+--
+-- The common parameters and defaults shared by all components
+-- of the replication system. It also provides default values
+-- for some critical parameters of the worker-side services.
+
+DROP TABLE IF EXISTS `config` ;
+
+CREATE TABLE IF NOT EXISTS `config` (
+
+  `category` VARCHAR(255) NOT NULL ,
+  `param`    VARCHAR(255) NOT NULL ,
+  `value`    VARCHAR(255) NOT NULL ,
+
+  UNIQUE  KEY (`category`,`param`)
+)
+ENGINE = InnoDB;
+
+-- -----------------------------------------------------
+-- Table `config_worker`
+-- -----------------------------------------------------
+--
+-- Worker-specific configuration parameters and overrides
+-- of the corresponidng default values if needed
+
+DROP TABLE IF EXISTS `config_worker` ;
+
+CREATE TABLE IF NOT EXISTS `config_worker` (
+
+  `name`         VARCHAR(255)       NOT NULL ,      -- the name of the worker
+
+  `is_enabled`   BOOLEAN            NOT NULL ,      -- is enabled for replication
+  `is_read_only` BOOLEAN            NOT NULL ,      -- a subclass of 'is_enabled' which restricts use of
+                                                    -- the worker for reading replicas. No new replicas can't be
+                                                    -- placed onto this class of workers.
+
+  `svc_host`     VARCHAR(255)       NOT NULL ,      -- the host name on which the worker server runs
+  `svc_port`     SMALLINT UNSIGNED  DEFAULT NULL ,  -- override for the global default
+
+  `fs_host`      VARCHAR(255)       NOT NULL ,      -- the host name on which the built-in FileServer runs
+  `fs_port`      SMALLINT UNSIGNED  DEFAULT NULL ,  -- override for the global default
+
+  `data_dir`     VARCHAR(255)       DEFAULT NULL ,  -- a file system path to the databases
+
+  PRIMARY KEY (`name`) ,
+
+  UNIQUE  KEY (`svc_host`, `svc_port`) ,
+  UNIQUE  KEY (`fs_host`,  `fs_port`)
+)
+ENGINE = InnoDB;
+
+
+-- --------------------------------------------------------------
+-- Table `config_worker_ext`
+-- --------------------------------------------------------------
+--
+-- The additional parameters overriding the defaults for individual
+-- worker services.
+
+DROP TABLE IF EXISTS `config_worker_ext` ;
+
+CREATE TABLE IF NOT EXISTS `config_worker_ext` (
+
+  `worker_name`  VARCHAR(255) NOT NULL ,
+  `param`        VARCHAR(255) NOT NULL ,
+  `value`        VARCHAR(255) NOT NULL ,
+
+  UNIQUE  KEY (`worker_name`, `param`, `value`) ,
```

Comment:
(`worker_name`, `param`) are not unique? I'd think worker_name + param should be PRIMARY KEY

Reply:
Agree